wiki/Gameplay/Judgement/osu!taiko/en.md

```diff
@@ -1,35 +1,55 @@
+
+Large notes may be hit with two keys of the correct colour at the same time (within less than 30 ms of each other) for double the score.
+
+### Drum rolls
+
+Drum rolls give 300 score (360 during Kiai time) while large drum rolls give 720 score (864 during Kiai time), per correctly timed drum roll tick.
+
+Hitting too quickly or too slowly will prevent the ticks from being collected. The bounds are roughly hitting twice as fast as ticks appear and hitting slower than every 5th tick.<!-- internal reference: https://github.com/peppy/osu-stable-reference/blob/1531237b63392e82c003c712faa028406073aa8f/osu!/GameplayElements/HitObjects/Taiko/SliderTaiko.cs#L362-L396 explanation is slightly simplified; bounds aren't exact because it calculates based on the time since the last hit tick's time, not since last button press -->
+
+With [ScoreV2](/wiki/Gameplay/Game_modifier/ScoreV2) enabled, drum rolls also give judgements depending on how many ticks are hit:<!-- internal reference: https://github.com/peppy/osu-stable-reference/blob/1531237b63392e82c003c712faa028406073aa8f/osu!/GameplayElements/HitObjects/Taiko/SliderTaiko.cs#L123-L144 -->
+
+| Judgement | Requirement |
+| GREAT | Ticks hit >= amount of ticks * (`0.3` if OD <= 6, otherwise `0.1 + OD / 30`) |
+| OK | At least one tick hit |
+| MISS | Anything else |
+
+### Swells
+
+Swells give 300 score per hit. Failing to complete the required amount of hits results in a [health](/wiki/Gameplay/Health) punishment, but they do not give judgements.
 
-- *GREAT* given if drummed inside the judgement circle perfectly with correct colour.
-- *GOOD* given if drummed slightly away from the judgement circle with correct colour.
-- *MISS/BAD* given if drummed too early, too late, missed the judgement circle, or drummed the wrong colour.
+With [ScoreV2](/wiki/Gameplay/Game_modifier/ScoreV2) enabled, swells give judgements depending on how many times they are hit:<!-- internal reference: https://github.com/peppy/osu-stable-reference/blob/1531237b63392e82c003c712faa028406073aa8f/osu!/GameplayElements/HitObjects/Taiko/SpinnerTaiko.cs#L151-L171 -->
 
-**Drumroll:**
+| Judgement | Required hits |
+| GREAT | 100% |
+| OK | 50% |
+| MISS | 0% |
 
-- Constant 300/600 score with Kiai Time bonus when available given per small notes drummed successfully in the drumroll with any colour.
-- No score given when drummed on the gap of the small notes.
-- No *MISS/BAD* for not doing the drumroll as per *Taiko no Tatsujin* style.
+## ScoreV2
 
-**Denden/Shaker:**
+The [ScoreV2](/wiki/Gameplay/Game_modifier/ScoreV2) mod changes a few things about osu!taiko judgement mechanics:
 
-- Constant 300 score given per hit/shake until completed.
-- *GREAT* given on the last hit/shake.
-- *MISS/BAD* damage given (not considered a miss) if failed to complete the Denden/Shaker successfully.
+- Drum roll drum speed restrictions are lifted.
```

Comment:
"Drum roll drum speed restrictions" is five nouns chained together, change them even so slightly for readability?

Reply:
```suggestion
- Speed restrictions for drum rolls are lifted, meaning they can be mashed without penalty.
```